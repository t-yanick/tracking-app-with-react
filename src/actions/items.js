const addItems = (items = []) => ({
  type: 'ADD_ITEMS',
  items,
});

export default addItems;