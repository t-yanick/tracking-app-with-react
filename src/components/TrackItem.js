import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import pluralize from 'pluralize';

const TrackItem = ({ item, result, targetDate }) => {
  const {
    icon, unit, title, target,
  } = item;
  const rate = result / target;
  const AchievementRate = Math.floor((rate >= 1 ? 1 : rate) * 100);

  return (
    <div className="items__item">
      <Link to={`/track/${Number(targetDate)}/edit`} className="items__item-link">
        <div className="items__icon">
          <span className="iconify" data-icon={icon || 'bi:pen-fill'} data-line="false" />
        </div>
        <div className="items__text">
          <div className="items__title">{title}</div>
          <div className="items__result">
            <span className="items__result__num">{result || '0'}</span>
            <span className="items__result__unit">{`${pluralize(unit, result)}/ ${target}`}</span>
          </div>
          <div className="items__rate">{`Achievement rate: ${AchievementRate || '0'}%`}</div>
        </div>
      </Link>
    </div>
  );
};

TrackItem.propTypes = {
  item: PropTypes.instanceOf(Object),
  result: PropTypes.number,
  targetDate: PropTypes.number,
};

TrackItem.defaultProps = {
  item: {},
  result: 0,
  targetDate: 0,
};

export default TrackItem;
