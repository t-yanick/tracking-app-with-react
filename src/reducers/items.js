const defaultItemsState = [];

const itemsReducer = (state = defaultItemsState, action) => {
  switch (action.type) {
    case 'ADD_ITEMS':
      return action.items;
    default:
      return state;
  }
};

export default itemsReducer;