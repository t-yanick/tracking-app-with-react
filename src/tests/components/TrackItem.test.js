import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import TrackItem from '../../components/TrackItem';
import createTestStore from '../fixtures/createTestStore';
import items from '../fixtures/items';

describe('TrackItem', () => {
  let store;
  beforeEach(() => {
    store = createTestStore();
  });
  test('should match with snapshot', () => {
    const handleSubmit = jest.fn();
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <TrackItem
              handleSubmit={handleSubmit}
              item={items[0]}
              result={100}
              targetDate={items[0].date}
            />
          </MemoryRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
