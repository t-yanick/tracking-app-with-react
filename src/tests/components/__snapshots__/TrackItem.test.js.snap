// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackItem should match with snapshot 1`] = `
<div
  className="items__item"
>
  <a
    className="items__item-link"
    href="/track/0/edit"
    onClick={[Function]}
  >
    <div
      className="items__icon"
    >
      <span
        className="iconify"
        data-icon="typcn:sort-alphabetically"
        data-line="false"
      />
    </div>
    <div
      className="items__text"
    >
      <div
        className="items__title"
      >
        Idioms
      </div>
      <div
        className="items__result"
      >
        <span
          className="items__result__num"
        >
          100
        </span>
        <span
          className="items__result__unit"
        >
          idioms/ 30
        </span>
      </div>
      <div
        className="items__rate"
      >
        Achievement rate: 100%
      </div>
    </div>
  </a>
</div>
`;
